Obsidian/Client.cs

```diff
@@ -25,106 +25,190 @@ namespace Obsidian;
 
-    public Server Server { get; private set; }
-    public Player? Player { get; private set; }
+    /// <summary>
+    /// The base network stream used by the <see cref="minecraftStream"/>.
+    /// </summary>
+    private readonly NetworkStream networkStream;
 
-    public ILogger Logger => Server.Logger;
+    /// <summary>
+    /// Used to continuously send and receive encrypted packets from the client.
+    /// </summary>
+    private readonly PacketCryptography packetCryptography;
 
-    public ConcurrentHashSet<(int X, int Z)> LoadedChunks { get; internal set; }
+    /// <summary>
+    /// The socket associated with the <see cref="networkStream"/>.
+    /// </summary>
+    private readonly Socket socket;
+
+    /// <summary>
+    /// The current server configuration.
+    /// </summary>
+    private readonly ServerConfiguration config;
 
     public Client(Socket socket, ServerConfiguration config, int playerId, Server originServer)
     {
         this.socket = socket;
         this.config = config;
         id = playerId;
-        packetCryptography = new PacketCryptography();
         Server = originServer;
-        LoadedChunks = new();
-        handler = new ClientHandler(config);
 
-        socketStream = new NetworkStream(socket);
-        minecraftStream = new MinecraftStream(socketStream);
+        LoadedChunks = new();
+        packetCryptography = new();
+        handler = new(config);
+        networkStream = new(socket);
+        minecraftStream = new(networkStream);
 
+        var linkOptions = new DataflowLinkOptions { PropagateCompletion = true };
         var blockOptions = new ExecutionDataflowBlockOptions { CancellationToken = cancellationSource.Token, EnsureOrdered = true };
-        packetQueue = new BufferBlock<IClientboundPacket>(blockOptions);
         var sendPacketBlock = new ActionBlock<IClientboundPacket>(packet =>
         {
             if (socket.Connected)
                 SendPacket(packet);
-        },
-        blockOptions);
+        }, blockOptions);
 
-        var linkOptions = new DataflowLinkOptions { PropagateCompletion = true };
-        packetQueue.LinkTo(sendPacketBlock, linkOptions);
+        packetQueue = new BufferBlock<IClientboundPacket>(blockOptions);
+        _ = packetQueue.LinkTo(sendPacketBlock, linkOptions);
 
         handler.RegisterHandlers();
     }
 
     private async Task<(int id, byte[] data)> GetNextPacketAsync()
     {
-        int length = await minecraftStream.ReadVarIntAsync();
-        byte[] receivedData = new byte[length];
+        var length = await minecraftStream.ReadVarIntAsync();
+        var receivedData = new byte[length];
 
-        await minecraftStream.ReadAsync(receivedData.AsMemory(0, length));
+        _ = await minecraftStream.ReadAsync(receivedData.AsMemory(0, length));
 
-        int packetId = 0;
-        byte[] packetData = Array.Empty<byte>();
+        var packetId = 0;
+        var packetData = Array.Empty<byte>();
```

Comment:
Not sure what we decided on code standards, I'm not sure whether we require explicit variable typing..

Reply:
I think its fine 🙏🏼